server/Forms.py

```diff
@@ -19,4 +20,32 @@ class SignUpForm(FlaskForm):
                         render_kw={"placeholder": "Email"})
     password = PasswordField('', 
                             validators=[InputRequired(), Length(min=6, max=80)],
-                            render_kw={"placeholder": "Password"})
\ No newline at end of file
+                            render_kw={"placeholder": "Password"})
+
+class TrainingForm(FlaskForm):
+    file = FileField('Reference image:', validators=[
+        FileRequired(),
+        FileAllowed(['jpg', 'jpeg'], 'Allowed file formats: .jpg, .jpeg!')
+    ])
+    personSelect = SelectMultipleField('Select an existing person', 
+                               choices=[
+                                 ('Kanye', 'Kanye'), 
+                                 ('Elon', 'Elon'), 
+                                 ('Kaden', 'Kaden'), 
+                                 ('Jacob', 'Jacob')
+                               ])
+    
+    newPersonName = StringField('Add a new person',
+                        validators=[Length(max=30)],
+                        render_kw={"placeholder": "New person name"})
+
+    def validate(self, extra_validators=None):
+        if super().validate(extra_validators):
+
+            if not ((self.personSelect.data and not self.newPersonName.data) or (not self.personSelect.data and self.newPersonName.data)):
+                self.newPersonName.errors.append('You must choose an existing person OR provide a new name, not both!')
+                return False
+            else:
+                return True
```

Comment:
This is a custom validation function. In this form, the user should **either** select an existing user OR enter a new name, but not **both**.